### PR TITLE
CRUD methods in todo list

### PR DESCRIPTION
**Pull request by sainingo**
# create, update, delete methods in a todo list

### CODE REVIEW

File: src/editTask.js [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+    editEl.forEach((edit) => {
+      edit.addEventListener('keypress', (e) => {
+        //  e.stopPropagation();
+        if (e.target.tagName === 'INPUT' && e.target.type === 'text' && e.key === 'Enter') {
+          updateText();
+        }
+      });
+    });
+  });
+}
```
Comment (keroloussamy) [inline src/editTask.js]: - [ ] Kindly your edit method works fine, but you should save the updated data to the local storage so when you reload the page you find the updated data.

image

File: src/index.js [hunk 2]
```diff
@@ -1,27 +1,46 @@
...
-const todoListContainer = document.querySelector('.todo-list');
+function updateUi(id) {
+  const localData = JSON.parse(localStorage.getItem('tasks'));
+  if (localData !== null) {
+    const deleteIndex = localData.findIndex((item) => item.index === id);
+    localData.splice(deleteIndex, 1);
+  }
+  newTask.tasks = localData;
+  localStorage.setItem('tasks', JSON.stringify(localData));
+}
```
Comment (keroloussamy) [inline src/index.js]: - [ ] Kindly in your delete method, when I try to delete the first item, it removes the last item from the local storage, kindly checks to remove the correct index from the local storage.

image

**Review by keroloussamy — CHANGES_REQUESTED** [review]
Hi @sainingo, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## To highlight 🎯

- No linter errors.✔️
- Good code design and implementation. ✔️

## Required Changes ♻️
_Check the comments under the review._



## Optional suggestions
_Check the comments under the review._

- [ ] x

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by ShoiraTa** [comment]
## **STATUS: APPROVED  🟢 ** 
------
### Hi @sainingo , 

**Great work on making the changes requested by a previous reviewer 👏🏻** 

Your project is complete! . There is nothing else to say other than... it's time to merge it 🍾🚢 :shipit: . Congratulations!  🎉💯🌟 

5SW

**Cheers and Happy coding! :clap: :clap: :clap:**

## OPTIONAL: 

- [ ] Deleting a task should update all remaining items' indexes, so they represent the current list order and are unique. So if you have task1 with id 1 and task2 with id 2, and delete task1, then task2 should get id 1. This requirement is optional because was not mentioned earlier. I strongly recommend implementing it

Feel free to leave any questions or comments in the PR thread by mentioning my Github username if something is not 100% clear.
***
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have 2 more limited reviews per this project. If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**